packages/unmock-core/src/service/matcher.ts

```diff
@@ -30,10 +38,43 @@ export class OASMatcher {
     );
     return reqPath.replace(regexToRemoveFromReqPath, "");
   }
+
+  private static buildRegexpForPaths(
+    schema: OpenAPIObject,
+  ): IEndpointToRegexMapping {
+    const mapping: IEndpointToRegexMapping = {};
+    const paths = schema.paths;
+    if (paths === undefined || paths.length === 0) {
+      return mapping;
+    }
+    Object.keys(paths).forEach((path: string) => {
+      const pathParameters = getPathParametersFromPath(path);
+      let newPath: string = "";
+      if (pathParameters.length === 0) {
+        newPath = path; // Simply convert to direct regexp pattern
+      } else {
+        const schemaParameters = getPathParametersFromSchema(paths, path);
+        newPath = buildPathRegexStringFromParameters(
+          path,
+          schemaParameters,
+          pathParameters,
+        );
+      }
+
+      const newPathRegex = XRegExp(`^${newPath}$`, "g");
+      mapping[path] = newPathRegex;
+    });
+    return mapping;
+  }
+
   private readonly schema: OpenAPIObject;
+  private readonly endpointToRegexMapping: IEndpointToRegexMapping = {};
```

Comment:
Removing initialization to `={}` here would ensure that it must be defined in the constructor in future as well.

Reply:
Nice catch, was a leftover from using `this.buildRegexpForPaths()` where typescript can't follow the logic to notice it's instantiated.